### PR TITLE
GitHub actions (tests and releasing)

### PR DESCRIPTION
**Pull request by bbelderbos**

### CODE REVIEW

File: .github/workflows/pre_release.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Publish to Test PyPI
+on:
+  push:
+    tags:
+      - '*.*.*'
+
+jobs:
+  test_pypi_release:
```
Comment (willfrey) [inline .github/workflows/pre_release.yml]:
```suggestion
  test-pypi-release:
```

File: .github/workflows/pre_release.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - name: Install Poetry
+        run: curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
```
Comment (willfrey) [inline .github/workflows/pre_release.yml:17]:
```suggestion
        run: curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -- -y
```

File: .github/workflows/pre_release.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  push:
+    tags:
+      - '*.*.*'
+
+jobs:
+  test_pypi_release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
```
Comment (willfrey) [inline .github/workflows/pre_release.yml:12]:
```suggestion
      - uses: actions/setup-python@v2
        with:
          python-version: 3.9
      
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+name: Publish to PyPI
+on:
+  release:
+    types: [published]
+
+jobs:
+  pypi_release:
```
Comment (willfrey) [inline .github/workflows/release.yml]:
```suggestion
  pypi-release:
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    name: Builds Using Poetry and Publishes to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - name: Install Poetry
+        run: curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
```
Comment (willfrey) [inline .github/workflows/release.yml:14]:
```suggestion
        run: curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/instal-poetry.py | python -- -y
```

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+on: push
+
+jobs:
+  tests:
+    name: Runs tests with Nox
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout repo content
+        uses: actions/checkout@v2
```
Comment (willfrey) [inline .github/workflows/tests.yml]:
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/tests.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+      - name: setup python
+        uses: actions/setup-python@v2
```
Comment (willfrey) [inline .github/workflows/tests.yml]:
```suggestion
      - uses: actions/setup-python@v2
        with:
          python-version: 3.9
```

File: .github/workflows/tests.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+      - name: install nox
```
Comment (willfrey) [inline .github/workflows/tests.yml]:
```suggestion
      - name: Install nox
```

File: .github/workflows/tests.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        run: pip install nox==2021.6.12
+
+      - name: install poetry
```
Comment (willfrey) [inline .github/workflows/tests.yml]:
```suggestion
      - name: Install poetry
```

File: .github/workflows/tests.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        run: pip install poetry==1.1.11
+
+      - name: install nox-poetry
```
Comment (willfrey) [inline .github/workflows/tests.yml]:
```suggestion
      - name: Install nox-poetry
```

File: .github/workflows/tests.yml [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        run: pip install nox-poetry==0.8.6
+
+      - name: run tests
```
Comment (willfrey) [inline .github/workflows/tests.yml]:
```suggestion
      - name: Run tests
```

**Review by willfrey — CHANGES_REQUESTED** [review]
Just requesting some clean-up and consistency changes.

File: .github/workflows/pre_release.yml (hunk 2 above)
Comment (bbelderbos) [reply .github/workflows/pre_release.yml:17]: This does not work for me:

```
$ curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -- -y
/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python: can't open file '-y': [Errno 2] No such file or directory
curl: (23) Failed writing body (0 != 1362)
```

File: .github/workflows/release.yml (hunk 5 above)
Comment (bbelderbos) [reply .github/workflows/release.yml:14]: ditto

File: .github/workflows/pre_release.yml (hunk 2 above)
Comment (pybites) [reply .github/workflows/pre_release.yml:17]: this fails on GH Actions as well: https://github.com/bbelderbos/biteme/runs/3813454151?check_suite_focus=true

**Review by willfrey — APPROVED** [review]

**Pull request merged**